### ISSUE TITLE
Sponge Compatibility issue

### ISSUE THREAD

**Issue by xxxLEOPARDxxx** [NerdHubMC/AnvilFix#5]
With the right click on the anvil, the interface does not open. And in the server gives such errors.

Forge 2822
Sponge 2822
Mine 1.12.2
Anvil 1.0.2

```
[19:25:15] [Server thread/ERROR] [FML]: Index: 9 Listeners:
[19:25:15] [Server thread/ERROR] [FML]: 0: HIGHEST
[19:25:15] [Server thread/ERROR] [FML]: 1: ASM: class tehnut.harvest.Harvest$EventHandler onInteract(Lnet/minecraftforge/event/entity/player/PlayerInteractEvent$RightClickBlock;)V
[19:25:15] [Server thread/ERROR] [FML]: 2: ASM: net.dries007.holoInventory.server.ServerEventHandler@6dd973b7 onRightClick(Lnet/minecraftforge/event/entity/player/PlayerInteractEvent$RightClickBlock;)V
[19:25:15] [Server thread/ERROR] [FML]: 3: ASM: com.attributestudios.wolfarmor.common.event.PlayerEventHandler@676bceb8 onPlayerInteract(Lnet/minecraftforge/event/entity/player/PlayerInteractEvent$RightClickBlock;)V
[19:25:15] [Server thread/ERROR] [FML]: 4: ASM: net.quetzi.morpheus.helpers.MorpheusEventHandler@6325f2e0 bedClicked(Lnet/minecraftforge/event/entity/player/PlayerInteractEvent$RightClickBlock;)V
[19:25:15] [Server thread/ERROR] [FML]: 5: HIGH
[19:25:15] [Server thread/ERROR] [FML]: 6: ASM: class com.feed_the_beast.ftbutilities.handlers.FTBUtilitiesPlayerEventHandler onRightClickBlock(Lnet/minecraftforge/event/entity/player/PlayerInteractEvent$RightClickBlock;)V
[19:25:15] [Server thread/ERROR] [FML]: 7: ASM: class techguns.events.TGEventHandler rightClickEvent(Lnet/minecraftforge/event/entity/player/PlayerInteractEvent$RightClickBlock;)V
[19:25:15] [Server thread/ERROR] [FML]: 8: NORMAL
[19:25:15] [Server thread/ERROR] [FML]: 9: ASM: class p455w0rd.anvilfix.init.ModEvents onRightClickBlock(Lnet/minecraftforge/event/entity/player/PlayerInteractEvent$RightClickBlock;)V
[19:25:15] [Server thread/ERROR] [FML]: 10: ASM: class com.cutievirus.creepingnether.CommonEvents usedItem(Lnet/minecraftforge/event/entity/player/PlayerInteractEvent$RightClickBlock;)V
[19:25:15] [Server thread/ERROR] [FML]: 11: ASM: class net.daveyx0.primitivemobs.core.PrimitiveMobsEvents$EntityEventHandler onBlockRightClickEvent(Lnet/minecraftforge/event/entity/player/PlayerInteractEvent$RightClickBlock;)V
[19:25:15] [Server thread/ERROR] [FML]: 12: ASM: biomesoplenty.common.handler.GrassPathEventHandler@3302cf3 onBlockRightClick(Lnet/minecraftforge/event/entity/player/PlayerInteractEvent$RightClickBlock;)V
[19:25:15] [Server thread/ERROR] [FML]: 13: ASM: crafttweaker.mc1120.events.CommonEventHandler@60cbbf3a onPlayerRightClickBlockEvent(Lnet/minecraftforge/event/entity/player/PlayerInteractEvent$RightClickBlock;)V
[19:25:15] [Server thread/ERROR] [FML]: 14: ASM: com.brandon3055.draconicevolution.handlers.DEEventHandler@2c4000b4 rightClickBlock(Lnet/minecraftforge/event/entity/player/PlayerInteractEvent$RightClickBlock;)V
[19:25:15] [Server thread/ERROR] [FML]: 15: ASM: com.grim3212.mc.pack.util.event.BlockChangeEvents@70186ef4 playerInteract(Lnet/minecraftforge/event/entity/player/PlayerInteractEvent$RightClickBlock;)V
[19:25:15] [Server thread/ERROR] [FML]: 16: ASM: com.zeitheron.hammercore.event.vanilla.TileHandler@7ebe0579 playerInteract(Lnet/minecraftforge/event/entity/player/PlayerInteractEvent$RightClickBlock;)V
[19:25:15] [Server thread/ERROR] [FML]: 17: ASM: net.torocraft.signedit.SignEdit@5fcbc37e editSign(Lnet/minecraftforge/event/entity/player/PlayerInteractEvent$RightClickBlock;)V
[19:25:15] [Server thread/ERROR] [FML]: 18: ASM: Block{minecraft:air} rightClickEvent(Lnet/minecraftforge/event/entity/player/PlayerInteractEvent$RightClickBlock;)V
[19:25:15] [Server thread/ERROR] [FML]: 19: ASM: com.flemmli97.improvedmobs.handler.EventHandlerAI@15e4bf60 openTile(Lnet/minecraftforge/event/entity/player/PlayerInteractEvent$RightClickBlock;)V
[19:25:15] [Server thread/ERROR] [FML]: 20: ASM: com.flemmli97.tenshilib.common.events.handler.CommonEvents@328592fe stopOffhand(Lnet/minecraftforge/event/entity/player/PlayerInteractEvent$RightClickBlock;)V
[19:25:15] [Server thread/ERROR] [FML]: 21: ASM: class com.cutievirus.creepingnether.CommonEvents onPlayerInteract(Lnet/minecraftforge/event/entity/player/PlayerInteractEvent;)V
[19:25:15] [Server thread/ERROR] [FML]: 22: ASM: crafttweaker.mc1120.events.CommonEventHandler@60cbbf3a onPlayerInteract(Lnet/minecraftforge/event/entity/player/PlayerInteractEvent;)V
[19:25:15] [Server thread/ERROR] [FML]: 23: ASM: treechopper.proxy.ServerProxy@591223f6 InteractWithTree(Lnet/minecraftforge/event/entity/player/PlayerInteractEvent;)V
[19:25:15] [Server thread/ERROR] [FML]: 24: ASM: com.flemmli97.improvedmobs.handler.EventHandlerAI@15e4bf60 pathDebug(Lnet/minecraftforge/event/entity/living/LivingEvent;)V
[19:25:15] [Server thread/ERROR] [FML]: 25: LOWEST
[19:25:15] [Server thread/ERROR] [FML]: 26: ASM: elec332.core.handler.event.PlayerEventHandler@2bb8485b onItemRightClick(Lnet/minecraftforge/event/entity/player/PlayerInteractEvent$RightClickBlock;)V
[19:25:15] [Server thread/FATAL] [net.minecraft.server.MinecraftServer]: Error executing task
java.util.concurrent.ExecutionException: java.lang.NoSuchMethodError: net.minecraft.inventory.ContainerRepair.<init>(Lnet/minecraft/entity/player/InventoryPlayer;Lnet/minecraft/world/World;Lnet/minecraft/entity/player/EntityPlayer;)V
	at java.util.concurrent.FutureTask.report(Unknown Source) ~[?:1.8.0_201]
	at java.util.concurrent.FutureTask.get(Unknown Source) ~[?:1.8.0_201]
	at net.minecraft.util.Util.func_181617_a(SourceFile:47) [h.class:?]
	at org.spongepowered.common.SpongeImplHooks.onUtilRunTask(SpongeImplHooks.java:297) [SpongeImplHooks.class:1.12.2-2768-7.1.5]
	at net.minecraft.server.MinecraftServer.redirect$onRun$zjm000(MinecraftServer.java:3963) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:723) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:397) [nz.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526) [MinecraftServer.class:?]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_201]
Caused by: java.lang.NoSuchMethodError: net.minecraft.inventory.ContainerRepair.<init>(Lnet/minecraft/entity/player/InventoryPlayer;Lnet/minecraft/world/World;Lnet/minecraft/entity/player/EntityPlayer;)V
	at p455w0rd.anvilfix.container.ContainerRepairHacked.<init>(ContainerRepairHacked.java:49) ~[ContainerRepairHacked.class:?]
	at p455w0rd.anvilfix.init.ModGuiHandler.getServerGuiElement(ModGuiHandler.java:28) ~[ModGuiHandler.class:?]
	at net.minecraftforge.fml.common.network.NetworkRegistry.redirect$setRemoteContainerPlugin$bcg000(NetworkRegistry.java:547) ~[NetworkRegistry.class:?]
	at net.minecraftforge.fml.common.network.NetworkRegistry.getRemoteGuiContainer(NetworkRegistry.java:253) ~[NetworkRegistry.class:?]
	at net.minecraftforge.fml.common.network.internal.FMLNetworkHandler.openGui(FMLNetworkHandler.java:88) ~[FMLNetworkHandler.class:?]
	at net.minecraft.entity.player.EntityPlayer.openGui(EntityPlayer.java:2550) ~[aed.class:?]
	at p455w0rd.anvilfix.init.ModEvents.onRightClickBlock(ModEvents.java:38) ~[ModEvents.class:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_4_ModEvents_onRightClickBlock_RightClickBlock.invoke(.dynamic) ~[?:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90) ~[ASMEventHandler.class:?]
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:747) ~[EventBus.class:?]
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:702) ~[EventBus.class:?]
	at org.spongepowered.mod.event.SpongeToForgeEventFactory.createAndPostPlayerInteractBlockEvent(SpongeToForgeEventFactory.java:441) ~[SpongeToForgeEventFactory.class:1.12.2-2768-7.1.5]
	at org.spongepowered.mod.event.SpongeToForgeEventFactory.createAndPostForgeEvent(SpongeToForgeEventFactory.java:253) ~[SpongeToForgeEventFactory.class:1.12.2-2768-7.1.5]
	at org.spongepowered.mod.event.SpongeModEventManager.post(SpongeModEventManager.java:338) ~[SpongeModEventManager.class:1.12.2-2768-7.1.5]
	at org.spongepowered.mod.event.SpongeModEventManager.extendedPost(SpongeModEventManager.java:435) ~[SpongeModEventManager.class:1.12.2-2768-7.1.5]
	at net.minecraft.server.management.PlayerInteractionManager.func_187251_a(PlayerInteractionManager.java:1229) ~[or.class:?]
	at net.minecraft.network.NetHandlerPlayServer.redirect$onProcessRightClickBlock$zip000(NetHandlerPlayServer.java:2417) ~[pa.class:?]
	at net.minecraft.network.NetHandlerPlayServer.func_184337_a(NetHandlerPlayServer.java:741) ~[pa.class:?]
	at net.minecraft.network.play.client.CPacketPlayerTryUseItemOnBlock.func_148833_a(SourceFile:55) ~[ma.class:?]
	at net.minecraft.network.play.client.CPacketPlayerTryUseItemOnBlock.func_148833_a(SourceFile:11) ~[ma.class:?]
	at org.spongepowered.common.event.tracking.phase.packet.PacketPhaseUtil.onProcessPacket(PacketPhaseUtil.java:193) ~[PacketPhaseUtil.class:1.12.2-2768-7.1.5]
	at net.minecraft.network.PacketThreadUtil$1.redirect$onProcessPacket$zli000(SourceFile:539) ~[hv$1.class:?]
	at net.minecraft.network.PacketThreadUtil$1.run(SourceFile:13) ~[hv$1.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[?:1.8.0_201]
	at java.util.concurrent.FutureTask.run(Unknown Source) ~[?:1.8.0_201]
	at net.minecraft.util.Util.func_181617_a(SourceFile:46) ~[h.class:?]
	... 7 more
[19:25:15] [Server thread/ERROR] [FML]: Exception caught during firing event net.minecraftforge.event.entity.player.PlayerInteractEvent$RightClickBlock@7263453c:
java.lang.NoSuchMethodError: net.minecraft.inventory.ContainerRepair.<init>(Lnet/minecraft/entity/player/InventoryPlayer;Lnet/minecraft/world/World;Lnet/minecraft/entity/player/EntityPlayer;)V
	at p455w0rd.anvilfix.container.ContainerRepairHacked.<init>(ContainerRepairHacked.java:49) ~[ContainerRepairHacked.class:?]
	at p455w0rd.anvilfix.init.ModGuiHandler.getServerGuiElement(ModGuiHandler.java:28) ~[ModGuiHandler.class:?]
	at net.minecraftforge.fml.common.network.NetworkRegistry.redirect$setRemoteContainerPlugin$bcg000(NetworkRegistry.java:547) ~[NetworkRegistry.class:?]
	at net.minecraftforge.fml.common.network.NetworkRegistry.getRemoteGuiContainer(NetworkRegistry.java:253) ~[NetworkRegistry.class:?]
	at net.minecraftforge.fml.common.network.internal.FMLNetworkHandler.openGui(FMLNetworkHandler.java:88) ~[FMLNetworkHandler.class:?]
	at net.minecraft.entity.player.EntityPlayer.openGui(EntityPlayer.java:2550) ~[aed.class:?]
	at p455w0rd.anvilfix.init.ModEvents.onRightClickBlock(ModEvents.java:38) ~[ModEvents.class:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_4_ModEvents_onRightClickBlock_RightClickBlock.invoke(.dynamic) ~[?:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90) ~[ASMEventHandler.class:?]
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:747) [EventBus.class:?]
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:702) [EventBus.class:?]
	at org.spongepowered.mod.event.SpongeToForgeEventFactory.createAndPostPlayerInteractBlockEvent(SpongeToForgeEventFactory.java:441) [SpongeToForgeEventFactory.class:1.12.2-2768-7.1.5]
	at org.spongepowered.mod.event.SpongeToForgeEventFactory.createAndPostForgeEvent(SpongeToForgeEventFactory.java:253) [SpongeToForgeEventFactory.class:1.12.2-2768-7.1.5]
	at org.spongepowered.mod.event.SpongeModEventManager.post(SpongeModEventManager.java:338) [SpongeModEventManager.class:1.12.2-2768-7.1.5]
	at org.spongepowered.mod.event.SpongeModEventManager.extendedPost(SpongeModEventManager.java:435) [SpongeModEventManager.class:1.12.2-2768-7.1.5]
	at net.minecraft.server.management.PlayerInteractionManager.func_187251_a(PlayerInteractionManager.java:1229) [or.class:?]
	at net.minecraft.network.NetHandlerPlayServer.redirect$onProcessRightClickBlock$zip000(NetHandlerPlayServer.java:2417) [pa.class:?]
	at net.minecraft.network.NetHandlerPlayServer.func_184337_a(NetHandlerPlayServer.java:741) [pa.class:?]
	at net.minecraft.network.play.client.CPacketPlayerTryUseItemOnBlock.func_148833_a(SourceFile:55) [ma.class:?]
	at net.minecraft.network.play.client.CPacketPlayerTryUseItemOnBlock.func_148833_a(SourceFile:11) [ma.class:?]
	at org.spongepowered.common.event.tracking.phase.packet.PacketPhaseUtil.onProcessPacket(PacketPhaseUtil.java:193) [PacketPhaseUtil.class:1.12.2-2768-7.1.5]
	at net.minecraft.network.PacketThreadUtil$1.redirect$onProcessPacket$zli000(SourceFile:539) [hv$1.class:?]
	at net.minecraft.network.PacketThreadUtil$1.run(SourceFile:13) [hv$1.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_201]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_201]
	at net.minecraft.util.Util.func_181617_a(SourceFile:46) [h.class:?]
	at org.spongepowered.common.SpongeImplHooks.onUtilRunTask(SpongeImplHooks.java:297) [SpongeImplHooks.class:1.12.2-2768-7.1.5]
	at net.minecraft.server.MinecraftServer.redirect$onRun$zjm000(MinecraftServer.java:3963) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:723) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:397) [nz.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526) [MinecraftServer.class:?]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_201]```

**Comment by p455w0rd** [comment]
RIP..thank you

**Comment by p455w0rd** [comment]
Fixed jar: https://minecraft.curseforge.com/projects/anvil-fix/files/2698663
